dask/distributed/tests/test_client.py

```diff
@@ -111,3 +113,9 @@ def test_register_collections():
         c.close()
         d.close()
 
+
+def test_register_with_scheduler():
+    with scheduler_and_workers() as (s, (a, b)):
+        c = Client(s.address_to_clients)
+        sleep(.2)
```

Comment:
```python
for i in range(100):
    if s.clients:
        break
    else:
        sleep(0.1)
```

A number of the intermittent errors are caused by `sleep`s.  I've been slowly removing them.

Alternatively the client could block until it receives an acknowledgment from the Scheduler.  This is probably cleaner.